MSRV for v6 is Rust 1.59

First of all, please read the [FAQ](https://github.com/Nukesor/comfy-table/blob/main/FAQ.md) if you're having styling issues!

Otherwise feel free to go ahead :)

#### Describe the bug

The `Cargo.toml` for this library states that the MSRV is 1.58, but it can't currently compile on 1.58

#### Expected behavior

Cloning the repo and building it should work on 1.58 or the MSRV should be declared as 1.59

#### Code

N/A

#### Steps to reproduce the bug

```
> git checkout v6.0.0-rc.1
> rustup install 1.58 # if not already installed
> cargo +1.58 build
```

Error:

```
error[E0658]: destructuring assignments are unstable
  --> src\utils\arrangement\dynamic.rs:62:46
   |
62 |         (remaining_width, remaining_columns) = enforce_lower_boundary_constraints(
   |         ------------------------------------ ^
   |         |
   |         cannot assign to this expression
   |
   = note: see issue #71126 <https://github.com/rust-lang/rust/issues/71126> for more information

error[E0658]: destructuring assignments are unstable
  --> src\utils\arrangement\dynamic.rs:85:46
   |
85 |         (remaining_width, remaining_columns) = optimize_space_after_split(
   |         ------------------------------------ ^
   |         |
   |         cannot assign to this expression
   |
   = note: see issue #71126 <https://github.com/rust-lang/rust/issues/71126> for more information

For more information about this error, try `rustc --explain E0658`.
error: could not compile `comfy-table` due to 2 previous errors
```

#### Additional context

- Operating System: Windows 10
- comfy-table version: v6.0.0-rc.1

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -27,7 +27,7 @@ jobs:
             - x86_64-pc-windows-msvc
             - x86_64-apple-darwin
             - wasm32-wasi
-        toolchain: [stable, nightly]
+        toolchain: [stable, nightly, 1.59]
         include:
           - target: x86_64-unknown-linux-gnu
             os: ubuntu-latest

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -9,7 +9,7 @@ documentation = "https://docs.rs/comfy-table/"
 license = "MIT"
 keywords = ["terminal", "table", "unicode"]
 readme = "README.md"
-rust-version = "1.58"
+rust-version = "1.59"
 edition = "2021"
 
 [badges]

